samples/99_genre_3d/04_ray_caster_advanced/app/main.rb

```diff
@@ -1,139 +1,166 @@
-=begin
-
-This sample is a more advanced example of raycasting that extends the previous 04_ray_caster sample.
-Refer to the prior sample to to understand the fundamental raycasting algorithm.
-This sample adds:
- * higher resolution of raycasting
- * Wall textures
- * Simple "drop off" lighting
- * Weapon firing
- * Drawing of sprites within the level.
+# This sample is a more advanced example of raycasting that is based on the lodev raycasting articles.
+# Refer to the prior sample to to understand the fundamental raycasting algorithm.
+# This sample adds:
+#  * Variable number of rays, field of view, and canvas size.
+#  * Wall textures
+#  * Inverse square law "drop off" lighting
+#  * Weapon firing
+#  * Drawing of sprites within the level.
 
 # Contributors outside of DragonRuby who also hold Copyright:
 # - James Stocks: https://github.com/james-stocks
 
-=end
-
 # https://github.com/BrennerLittle/DragonRubyRaycast
+# https://lodev.org/cgtutor/raycasting.html
 # https://github.com/3DSage/OpenGL-Raycaster_v1
```

Comment:
We could consider cutting the 3DSage refs since we're now following lodev. Actually that would help users understand that the basic sample follows 3DSage, and this one follows lodev.